databricks/koalas/indexes.py

```diff
@@ -587,6 +587,66 @@ def copy(self, name=None):
             result.name = name
         return result
 
+    def droplevel(self, level):
+        """
+        Return index with requested level(s) removed.
+        If resulting index has only 1 level left, the result will be
+        of Index type, not MultiIndex.
+
+        Parameters
+        ----------
+        level : int, str, or list-like, default 0
+            If a string is given, must be the name of a level
+            If list-like, elements must be names or indexes of levels.
+
+        Returns
+        -------
+        Index or MultiIndex
+
+
+        Examples
+        --------
+        >>> midx = ks.DataFrame({'a': ['a', 'b']}, index=[['a', 'x'], ['b', 'y'], [1, 2]]).index
+        >>> midx  # doctest: +SKIP
+        MultiIndex([('a', 'b', 1),
+                    ('x', 'y', 2),
+                   )
+
+        >>> midx.droplevel([0, 1])
+        Int64Index([1, 2], dtype='int64')
+
+        >>> midx.droplevel(0) # doctest: +SKIP
+        MultiIndex([('b', 1),
+                    ('y', 2)]
+                   )
+        """
+        if not isinstance(level, (tuple, list)):
+            level = [level]
+
+        for l in level:
+            if isinstance(l, int) and (l > self.nlevels - 1):
+                raise IndexError("Too many levels: Index has only {} levels, not {}"
+                                 .format(self.nlevels, l+1))
+            if isinstance(l, str) and (l not in self._internal.index_columns):
```

Comment:
`self._internal.index_names` instead of `self._internal.index_columns`?

```py
>>> midx = ks.DataFrame({'a': ['a', 'b']}, index=[['a', 'x'], ['b', 'y'], [1, 2]]).index
>>> midx.names = ['l1', 'l2', 'l3']
>>> midx
MultiIndex(levels=[['a', 'x'], ['b', 'y'], [1, 2]],
           codes=[[0, 1], [0, 1], [0, 1]],
           names=['l1', 'l2', 'l3'])
>>> midx.droplevel('l1')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/ueshin/workspace/databricks-koalas/worktrees/work/databricks/koalas/indexes.py", line 631, in droplevel
    raise KeyError("Level {} not found".format(l))
KeyError: 'Level l1 not found'
```